src/Components/Components/src/Routing/Router.cs

```diff
@@ -114,8 +127,21 @@ private static string StringUntilAny(string str, char[] chars)
                 : str.Substring(0, firstIndex);
         }
 
+        private void RefreshRouteTable()
+        {
+            var assemblies = AdditionalAssemblies == null ? new[] { AppAssembly } : new[] { AppAssembly }.Concat(AdditionalAssemblies);
+            var assembliesSet = new HashSet<Assembly>(assemblies);
+            if (_assemblies.Count != assembliesSet.Count)
```

Comment:
If the count matches you still have to compare the two sets of assemblies since the user might have replaced an assembly with a different one, which make the count equal but the list of assemblies different.

Reply:
Dupe of comment above